src/appleseed/foundation/image/colormap.h

```diff
@@ -0,0 +1,110 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2019 Achal Pandey, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once
+
+// appleseed.foundation headers.
+#include "foundation/image/color.h"
+#include "foundation/image/image.h"
+#include "foundation/image/tile.h"
+#include "foundation/math/aabb.h"
+
+// Standard headers.
+#include <vector>
+
+namespace foundation
+{
+
+class ColorMap
+{
+  public:
+  	static void find_min_max_red_channel(
+		const Image*    image,
+		const AABB2u&   crop_window,
+		float&          min_val,
+		float&          max_val);
+
+	static void find_min_max_relative_luminance(
```

Comment:
This method should also take a crop window. We want the API to be as orthogonal as possible, especially in appleseed.foundation, even if we don't happen to use the crop window when we remap luminance (why?).

Reply:
`remap_relative_luminance` should use the crop window. The whole frame was used in `ColorMapPostProcessingStage` so I didn't think about it much.

When I used the whole frame instead of just the crop window and selected a specific render region, everything outside that region which is supposed to be black turned purple-ish (in case of Inferno color map) which I believe is not the intended behavior. However, using the crop window fixes this which obviously makes sense and I think should be the correct behavior all along.